packages/bs-ethereum/src/BitqueryEDSEthereum.ts

```diff
@@ -0,0 +1,61 @@
+import { Currency, ExchangeDataService, NetworkType, TokenPricesResponse } from '@cityofzion/blockchain-service'
+import { Client, cacheExchange, fetchExchange, gql } from '@urql/core'
+import fetch from 'node-fetch'
+import { BITQUERY_API_KEY, BITQUERY_URL } from './constants'
+import dayjs from 'dayjs'
+import utc from 'dayjs/plugin/utc'
+import { bitqueryGetPricesQuery } from './graphql'
+
+dayjs.extend(utc)
+export class BitqueryEDSEthereum implements ExchangeDataService {
+  private readonly client: Client
+  private readonly networkType: NetworkType
+
+  constructor(networkType: NetworkType) {
+    this.networkType = networkType
+
+    this.client = new Client({
+      url: BITQUERY_URL,
+      exchanges: [cacheExchange, fetchExchange],
+      fetch,
+      fetchOptions: {
+        headers: {
+          'X-API-KEY': BITQUERY_API_KEY,
+        },
+      },
+    })
+  }
+
+  async getTokenPrices(currency: Currency): Promise<TokenPricesResponse[]> {
```

Comment:
I think this could be cleaner by breaking out into a helper method:

```typescript
  async getTokenPrices(currency: Currency): Promise<TokenPricesResponse[]> {
    if (this.networkType !== 'mainnet') throw new Error('Exchange is only available on mainnet');

    const after = dayjs.utc().subtract(2, 'day').startOf('date').toISOString();

    const result = await this.client.query(bitqueryGetPricesQuery, { after, network: 'ethereum' }).toPromise();
    if (result.error) throw new Error(result.error.message);
    if (!result.data) throw new Error('There is no price data');

    const currencyRatio = currency === 'USD' ? 1 : await getCurrencyRatio(currency);

    const prices = result.data.ethereum.dexTrades.map((trade) => ({
      symbol: trade.baseCurrency.symbol,
      amount: trade.quotePrice * currencyRatio,
    }));

    return prices;
  }
}
```

```typescript
async function getCurrencyRatio(currency: Currency): Promise<number> {
  const request = await fetch(`https://api.flamingo.finance/fiat/exchange-rate?pair=USD_${currency}`, {
    method: 'GET',
  });
  const data = await request.json();
  return data;
}
```